src/openrct2/actions/StaffSetPatrolAreaAction.cpp

```diff
@@ -40,11 +48,6 @@ GameActions::Result StaffSetPatrolAreaAction::Query() const
         return GameActions::Result(GameActions::Status::InvalidParameters, STR_NONE, STR_NONE);
     }
 
-    if (!LocationValid(_loc))
```

Comment:
you still want to run through LocationValid on all locations. Also don't you need to truncate the range if it goes off map?

Reply:
Are there helper methods for MapRange that I can use?

I don't want to deny the action if just one tile is invalid. Any invalid tiles could be ignored by the Staff / PatrolArea methods.

Reply:
The location valid is meant to not just check if the location is inside bounds but also pass checking to plugins to allow them to block locations. So you basically want to break it up into an individual tile loop. This is done in the other tools that effect an area as well making it only remove those that are not valid